src/lazy/binary/raw/v1_1/immutable_buffer.rs

```diff
@@ -882,34 +885,51 @@ impl<'a> BinaryBuffer<'a> {
                 let bias = ((opcode.byte as usize & 0x0F) << 16) + 4160;
                 let (fixed_uint, input_after_opcode) = self.consume(1).read_fixed_uint(2)?;
                 let address = fixed_uint.value().expect_usize()? + bias;
-                (address, input_after_opcode)
+                (MacroIdRef::LocalAddress(address), input_after_opcode)
             }
             // Length-prefixed is a special case.
             EExpressionWithLengthPrefix => return self.read_eexp_with_length_prefix(opcode),
+            SystemEExpression => {
```

Comment:
🗺️ The actual read logic for binary e-expressions. This is the only arm that returns something other than a `MacroIdRef::LocalAddress`—it returns a `MacroIdRef::SystemAddress`.